docs/src/main/sphinx/release/release-412.md

```diff
@@ -0,0 +1,47 @@
+# Release 412 (5 Apr 2023)
+
+## General
+
+* Add statistics for optimizer rules to query statistics. The number of rules
+  statistics are collected for can be limited with the
+  `query.reported-rule-stats-limit` configuration property. ({issue}`2578`)
+* Add support for aggregate functions and parameters as arguments for the
+  [`json_object()`](json_object) and [`json_array()`](json_array)
+  functions. ({issue}`16489`, {issue}`16523`, {issue}`16525`)
+* Add {func}`exclude_columns` table function. ({issue}`16584`)
+* Allow table functions to return anonymous columns. ({issue}`16584`)
+* Allow disabling use of the cost-based optimizer for determining partitioning
+  of a stage with the `optimizer.use-cost-based-partitioning` configuration
+  property or the `use_cost_based_partitioning` session property. ({issue}`16781`)
+* Improve performance of queries involving table functions with table arguments. ({issue}`16012`)
+* Improve latency for small queries when fault-tolerant execution is enabled. ({issue}`16103`)
+
+## JDBC driver
+
+* Allow configuring a custom DNS resolver. ({issue}`16647`)
+
+## Delta Lake connector
+
+* Add support for arithmetic binary expressions in check constraints. ({issue}`16721`)
+* Improve performance of queries that only read partition columns. ({issue}`16788`)
+
+## Hive connector
+
+* Fix query failure when bucketing or sorting columns are registered in
+  a metastore in uppercase. ({issue}`16796`)
```

Comment:
sorry .. whats in uppercase .. the metastore? the columns? .. I think we need to reword this